packages/core/useBattery/index.ts

```diff
@@ -26,7 +26,7 @@ export function useBattery(options: ConfigurableNavigator = {}) {
   const { navigator = defaultNavigator } = options
   const events = ['chargingchange', 'chargingtimechange', 'dischargingtimechange', 'levelchange']
 
-  const isSupported = useSupported(() => navigator && 'getBattery' in navigator)
+  const isSupported = useSupported(() => navigator && typeof navigator.getBattery === 'function')
```

Comment:
```suggestion
  const isSupported = useSupported(() => navigator && 'getBattery' in navigator && typeof navigator.getBattery === 'function')
```

Reply:
I fixed it even though I didn't like it. Can we approach it in ts style instead of js too? I mean it's redundant to check an unimportant condition
```ts

  const isSupported = useSupported(() => navigator && typeof (navigator as Navigator & {
    getBattery?: () => unknown
  }).getBattery === 'function')
```